### PR TITLE
Feature/display movie spec

### PR DESCRIPTION
**Pull request by dustingouner**
Files changed: app.js, movieCard.js, MovieDatails.js, Movies.js, movie-details.css
Change Summary: Refactored styling for individual movie page, added propTypes to components, back to home button functionality, added server error handling in app.js
Additional Questions/Concerns: Need to revisit error handling to make sure all 500/400 errors are being handled properly. Will need to continue working on styling for individual movie page and media scroller.

### CODE REVIEW

**Comment by SamanthaMcElhinney** [comment]
This looks great. Nice job with the logic for rendering to the home page. I'll work on the error handling today for 500 errors and styling for the home page.

File: src/App.js [hunk 1]
```diff
@@ -40,8 +41,17 @@ class App extends Component {
...
+    .then((response) => {
+      if(!response.ok) {
+        this.setState({error: 'server error'})
+        throw Error(response.status)
+      } else {
+        return response.json()
+      }
+    })
     .then((data) => this.setState({movies: data.movies, individualMovie: null}))
+    .catch(error => alert(`error at ${error}`))
```
Comment (niksseif) [inline src/App.js:53]: Assigning 'null' to `individualMovie` might create some bugs in your code. The best practice is always to assign the data type that you're expecting to receive from the API. Is `individualMovie` an object, array, number, or string?

File: src/Movies.js [hunk 2]
```diff
@@ -21,4 +22,9 @@ function Movies({ movies, getMovieInfo }) {
 }
 
 
-export default Movies
\ No newline at end of file
+export default Movies
+
+Movies.propTypes = {
```
Comment (niksseif) [inline src/Movies.js:27]: Nice work 🙌🏼

File: src/App.js [hunk 3]
```diff
@@ -20,7 +20,12 @@ class App extends Component {
   getMovieInfo = (id) => {
     fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
     .then((response) => response.json())
-    .then((data) => this.setState({individualMovie: data, movies:null}))
+    .then((data) => this.setState({individualMovie: data}))
+  }
+
+  displayHome = () => {
+    this.setState({individualMovie: null})
```
Comment (niksseif) [inline src/App.js:28]: Follow my other comments about using null within states.

File: src/App.js [hunk 4]
```diff
@@ -20,7 +20,12 @@ class App extends Component {
   getMovieInfo = (id) => {
     fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
     .then((response) => response.json())
```
Comment (niksseif) [inline src/App.js:23]: What happens if we get 4xx errors here?

File: src/App.js [hunk 5]
```diff
@@ -13,6 +13,7 @@ class App extends Component {
     this.state = {
       movies: [],
       individualMovie: null,
```
Comment (niksseif) [inline src/App.js:15]: Assigning 'null' to 'individualMovie' might create some bugs in your code. The best practice is always to assign the data type that you're expecting to receive from the API. Is 'individualMovie' an object, array, number, or string?

File: src/App.js [hunk 6]
```diff
@@ -40,8 +41,17 @@ class App extends Component {
 
   componentDidMount() {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-    .then((response) => response.json())
+    .then((response) => {
+      if(!response.ok) {
+        this.setState({error: 'server error'})
+        throw Error(response.status)
```
Comment (niksseif) [inline src/App.js:47]: Consider moving fetch and the 'throw Error' into the API call within Apicall.js to adhere to the DRY principle. Please let me know if you have any questions about this. In this context, you can use the catch block to handle any errors that are returned from the API.

**Review by niksseif — COMMENTED** [review]
I left you few suggestions. Please reach out if you have any questions.

**Pull request merged**